specs/separate_configuration_files_per_component.adoc

```diff
@@ -0,0 +1,234 @@
+= Specification for segregating configurations per module wise
+
+This specification describes, Currently all the configuration details of
+different modules are read from same file. Instead of that store the
+configuration details in different files as well as etcd and read from
+particular source based on priority.
+
+== Problem description
+
+Currently, All the tendrl modules are configured from same file. Also, while
+import flow the configuration file gets appended with SDS specific
+configurations while installation of SDS integration bits on the storage nodes.
+To avoid this we have to store the configuration details into layer wise.
+
+1. Store the configuration details into different files as per modules, And
+   configure modules using those files. (high priority).
+2. Each modules should store its own configuration details to etcd and configured
+   via api request (Less priority).
+
+Only node_agent configuration file have its own configuration details and
+etcd configuration details.
+
+At the time of installation, Each module's configuration details should store in
+different files with proper name. Always tendrl modules should configured from
+those files only.
+When module run the configuration details should be stored in etcd.
+
+* Integration modules should store its configuration details under cluster
+  specific hierarchy in the central store.
+* Other modules should store its configuration with proper hierarchy.
+
+The configuration in the central store needs to be treated as an object
+hierarchy, which appropriate atoms and flows to allow modification of this
+object hierarchy. If any modification on configurations in etcd, Then it should
+affect a local configuration files also. But any changes in local configuration
+file should not affect configurations in etcd. Any changes in integration
+specific configuration in etcd will affect config files of particular cluster
+nodes only. But changes in other configuration in etc will affect all nodes
+local config files.
+
+== Use Cases
+
+Centralized deployment-wide or cluster-wide configuration in the central store,
+which can be overridden per component via configuration files or command line
+options.
```

Comment:
About "there will be Rest API functions for CRUD for options in Etcd"
This has not been implemented yet, and IMO this should be on a very low priority (in terms of our timelines) since configuration management is already solved by puppet/ansible etc per node 

Questions
1) What will store common options into Etcd during installation?
Ans: Nobody, config is required and pushed to etcd by each tendrl service on start

Once we have the API for conifgs:
2) What will happen during potential update of options in Etcd and local files?
Ans: Etcd config will be the only source of config, however during update the node local config changes will be overwritten on the etcd config

3) Will API be able to change options stored in local files?
Ans: Yes

4)What will be preferred way to configure Tendrl components on nodes in Ceph/Gluster cluster? For example I would like to increase logging level for group of nodes
Ans: Before tendrl services have started, use node local config files
Once tendrl services have started and updated config in etcd, use tendrl API to configure tendrl nodes